### PR TITLE
Tkachenko/Statistics by city added

### PR DESCRIPTION
**Pull request by fkma13552**

### CODE REVIEW

**Review by DmyMi — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi/Controllers/V1/StatisticController.cs [hunk 1]
```diff
@@ -59,17 +63,20 @@ public async Task<IActionResult> GetDirections(int limit)
...
         /// <param name="limit">The number of entries.</param>
+        /// <param name="city">City to look for.</param>
         /// <returns>List of popular workshops.</returns>
         [HttpGet]
         [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<WorkshopDTO>))]
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         [AllowAnonymous]
-        public async Task<IActionResult> GetWorkshops(int limit)
+        public async Task<IActionResult> GetWorkshops(int limit, [FromQuery] string city = null)
```
Comment (Bogdan-Hasanov) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/V1/StatisticController.cs:73]: string is nullable, so we don't need to set it manually

**Review by Bogdan-Hasanov — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/StatisticControllerTests.cs [hunk 2]
```diff
@@ -30,60 +30,72 @@ public void SetUp()
...
 
         [Test]
-        [TestCase(5)]
-        [TestCase(2)]
-        [TestCase(12)]
-        public async Task GetWorkshops_WhenLimitIsValid_ShouldReturnOkResultObject(int limit)
+        [TestCase(5, "")]
+        [TestCase(2, "")]
+        [TestCase(12, "")]
+        public async Task GetWorkshops_WhenLimitIsValid_NoCity_ShouldReturnOkResultObject(int limit, string city)
```
Comment (h4wk13) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/StatisticControllerTests.cs:36]: What does `NoCity` actually means ?
also `LimitIsValid` means that the  value is from the valid value range ?
If you want to use the value from your existing data - use collection you're set up for that case.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/StatisticControllerTests.cs [hunk 3]
```diff
@@ -30,60 +30,72 @@ public void SetUp()
...
         {
             // Arrange
-            service.Setup(s => s.GetPopularWorkshops(It.IsInRange(3, 10, Range.Inclusive))).ReturnsAsync(workshops);
+            service.Setup(s =>
```
Comment (h4wk13) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/StatisticControllerTests.cs:39]: could you please give that `service` more meaningful name ?

File: OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/StatisticControllerTests.cs [hunk 4]
```diff
@@ -30,60 +30,72 @@ public void SetUp()
         }
 
         [Test]
```
Comment (h4wk13) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/StatisticControllerTests.cs:32]: Please remove that attribute

**Pull request merged**